### DISCUSSION TITLE
Reproducing the same images

### DISCUSSION THREAD

**Discussion by SulaimanHasani** [facebookresearch/fastMRI#226] [Q&A]
Hi,
I have trained a unet using the code unet_train_demo and the data set available from FastMRI, When comparing the output images from running the trained unet with the output images from running the pre trained model available online unet_knee_sc_leaderboard I don't get the same quality of images. 

Can you please guide me on how to reproduce the same quality of images. 

Thank you for your support

**Comment by mmuckley** [comment]
Converting to a Discussion as this does not seemingly document a bug.

Hello @SulaimanHasani, the demo script is a smaller U-Net intended to help new people get started with the software. For the leaderboard submission, you can see the results in [this folder](https://github.com/facebookresearch/fastMRI/tree/main/fastmri_examples/unet/unet_reproduce_20201111.py).

**Comment by SulaimanHasani** [comment]
Thank you @mmuckley  for this information, I ran the code in the folder that you have shared with me and the results I get is shown below, the left image is the input image where the right image is the output from the U-net. 

![inputvsoutput](https://user-images.githubusercontent.com/100658813/158026262-dca79e60-a7a5-455d-aa5b-f30221ce876e.png)

Not sure what I am doing wrong.

**Reply by mmuckley** [reply]
@SulaimanHasani is this from running the test loop with PyTorch Lightning? There are some issues for loading model weights with PyTorch Lightning when running the test loop, depending on what version of PyTorch Lightning you have. I don't know the exact issue, but basically PyTorch Lightning may not load your model weights at all from a `resume_from_checkpoint` if you just run the `test`.

You should be able to run this script with the pretrained model and get good results https://github.com/facebookresearch/fastMRI/blob/main/fastmri_examples/unet/run_pretrained_unet_inference.py.

**Comment by SulaimanHasani** [comment]
Dear @mmuckley , 

I really appreciate your reply and the explanation of the situation, Just to clarify, I have no issue when running the run_pretrained_unet_inference.py using the pretrained model, it does give me good results, however when I generate a model using the unet_knee_sc_leaderboard.py the results I get when running run_pretrained_unet_inference.py and using this generated model is as shown in the below:
the left image is the input image where the right image is the output from the U-net.
![inputvsoutput](https://user-images.githubusercontent.com/100658813/158938894-b47b0e2a-2e39-482c-929d-bdfff837e022.png)

**Reply by mmuckley** [reply]
Hello @SulaimanHasani, in that case I do suspect an issue with PyTorch Lightning. Have you tried using the `load_state_dict` from a `torch.nn.Module` directly rather than loading the weights via PyTorch Lightning? It requires you to write a little bit of code to match the `dict` keys.
